kernel/thread.c

```diff
@@ -290,9 +324,40 @@ k_tid_t k_thread_create(struct k_thread *new_thread,
 	}
 	return new_thread;
 }
-#endif
 
-int k_thread_cancel(k_tid_t tid)
+
+#ifdef CONFIG_USERSPACE
+u32_t _handler_k_thread_create(u32_t new_thread, u32_t stack_p,
+			       u32_t stack_size, u32_t entry,
+			       u32_t p1, u32_t arg6, void *ssf)
+{
+	struct _syscall_10_args *margs = (struct _syscall_10_args *)arg6;
+	k_thread_stack_t stack = (k_thread_stack_t)stack_p;
+	u32_t guard_size;
+
+	_SYSCALL_IS_OBJ(new_thread, K_OBJ_THREAD, 1, ssf);
+	_SYSCALL_MEMORY(margs, sizeof(*margs), 0, ssf);
+
+	/* Thread needs to have rw access to the entire stack region including
+	 * any guard areas
+	 */
+	guard_size = (u32_t)K_THREAD_STACK_BUFFER(stack) - (u32_t)stack;
+	_SYSCALL_MEMORY(stack, stack_size + guard_size, 1, ssf);
```

Comment:
stacks are supposed to be opaque so maybe better to add an API to get the full size instead of calculating here